### PR TITLE
Update Visual Studio signatures in vamp/msvc/__init__.py

### PR DESCRIPTION
**Pull request by upside2**
Organized these into 4 groups:
A - SEH:  added another seh4_epilog signature generated by VS 2015 and VS 2017.
B - alloca_probe:  renamed an existing signature from gs_prolog so the emulator will correctly modify stack size.
C - security_check_cookie:  added another signature generated by 32-bit VS 2015 and VS 2017.  Added two signatures for 64-bit call, one in VS versions prior to 2015, the other for VS 2015 and 2017.  Removed a signature that is never seen.
D - report_gsfailure:  no changes.

Visual Studio adds "bnd" opcodes in the C-runtime beginning with VS 2015.  This explains most of the changes.

### CODE REVIEW

File: envi/__init__.py [hunk 1]
```diff
@@ -1186,21 +1186,23 @@ def executeCall(self, emu, va, args=None, ra=None):
         self.setupCall(emu, args=args, ra=ra)
         emu.setProgramCounter(va)
 
-    def execCallReturn(self, emu, value, argc):
+    def execCallReturn(self, emu, value, argc, apiReturnsVoid=False):
```
Comment (atlas0fd00m) [inline envi/__init__.py]: please keep this in the other PR :)  212

**Comment by atlas0fd00m** [comment]
@upside2 i apologize, i think i missed the value you were providing here.  i'm re-reviewing.

File: vivisect/impapi/windows/rpcrt4_32.py [hunk 2]
```diff
@@ -0,0 +1,522 @@
...
+    'rpcrt4.pfnmarshallroutines':( 'int', None, 'cdecl', 'rpcrt4.pfnMarshallRoutines', () ),
+    'rpcrt4.pfnsizeroutines':( 'int', None, 'cdecl', 'rpcrt4.pfnSizeRoutines', () ),
+    'rpcrt4.pfnunmarshallroutines':( 'int', None, 'cdecl', 'rpcrt4.pfnUnmarshallRoutines', () ),
+    'rpcrt4.short_array_from_ndr':( 'int', None, 'stdcall', 'rpcrt4.short_array_from_ndr', (('int', None), ('int', None), ('int', None), ('int', None)) ),
+    'rpcrt4.short_from_ndr':( 'int', None, 'stdcall', 'rpcrt4.short_from_ndr', (('int', None), ('int', None)) ),
+    'rpcrt4.short_from_ndr_temp':( 'int', None, 'stdcall', 'rpcrt4.short_from_ndr_temp', (('int', None), ('int', None), ('int', None)) ),
+    'rpcrt4.tree_into_ndr':( 'int', None, 'stdcall', 'rpcrt4.tree_into_ndr', (('int', None), ('int', None), ('int', None), ('int', None)) ),
+    'rpcrt4.tree_peek_ndr':( 'int', None, 'stdcall', 'rpcrt4.tree_peek_ndr', (('int', None), ('int', None), ('int', None), ('int', None)) ),
+    'rpcrt4.tree_size_ndr':( 'int', None, 'stdcall', 'rpcrt4.tree_size_ndr', (('int', None), ('int', None), ('int', None), ('int', None)) ),
+    # FIXME unresolved I_RpcAbortAsyncCall -> RPCRT4.I_RpcAsyncAbortCall
```
Comment (atlas0fd00m) [inline vivisect/impapi/windows/rpcrt4_32.py:511]: @upside2 , what do these FIXME's mean?

Comment (upside2) [reply vivisect/impapi/windows/rpcrt4_32.py:511]: I do not know.  The comments go back 6 years to when our current repository was started.  They were in the original impapi/windows/i386.py file, probably from Visi or Repete.

Comment (atlas0fd00m) [reply vivisect/impapi/windows/rpcrt4_32.py:511]: thanks.  again, please accept my apologies for missing the core value of this PR and letting it drop.

this PR currently can't be merged due to an upstream conflict with master.  

the conflict is simple but would you mind pulling in the latest master and fixing the conflict?  
i've done it on my end, but i don't believe i have rights to push to your branch.

**Comment by atlas0fd00m** [comment]
this PR currently can't be merged due to an upstream conflict with master.

the conflict is simple but would you mind pulling in the latest master and fixing the conflict?
i've done it on my end, but i don't believe i have rights to push to your branch.

File: vivisect/impapi/windows/rpcrt4_32.py (hunk 2 above)
Comment (upside2) [reply vivisect/impapi/windows/rpcrt4_32.py:511]: Yes, I will get to it next week, I hope

Comment (upside2) [reply vivisect/impapi/windows/rpcrt4_32.py:511]: I submitted a new PR 566 to replace this.  I think that will be easier.

Comment (atlas0fd00m) [reply vivisect/impapi/windows/rpcrt4_32.py:511]: for VAMP sigs, yes that was easier.  i just merged #566.

for Import API stuff alone, this PR still has a lot of value.  i really like how you've broken out the many areas of impapi sections, and support more msvc* libraries.
also, your changes before were cleaner and prettier than what we had for MSVS Vamp.

i really hated adding in whether a function returned void, but after reviewing it again i understand why.  my mindset when you first submitted it was in heavy full-emulation, so it didn't click with me.  at that point i was wildly overwhelmed, and i recoiled (i apologize).  looking at how you used it in impapi for emulation and symboliks, i get it now.

if you are interested, this PR represents a lot of work, and i'd like to merge it in and have you get credit.  i may create a PR separately if you don't want to, and @rakuy0 will have to review it.

but if you want the win, it doesn't seem like a lot of work to even create a new branch and copy over the work from this one, if the merge-collisions are too annoying.

either way, thank you for your contribution.

Comment (upside2) [reply vivisect/impapi/windows/rpcrt4_32.py:511]: Thanks for the kind words!  I will separate the two logical sections as new branches; this original PR had too many things thrown into it.  I will also be sending a couple other fixes soon.

**Pull request closed without merging**